lib/models/device.coffee

```diff
@@ -275,23 +275,20 @@ exports.identify = (uuid, callback) ->
 # @param {String} newName - the device new name
 # @param {module:resin.models.device~renameCallback} callback - callback
 #
-# @todo This action doesn't return any error
-# if trying to rename a device that does not
-# exists. This should be fixed server side.
-#
 # @example
 #	resin.models.device.rename 317, 'NewName', (error) ->
 #		throw error if error?
 #		console.log("Device has been renamed!")
 ###
 exports.rename = (name, newName, callback) ->
-	return pine.patch
-		resource: 'device'
-		body:
-			name: newName
-		options:
-			filter:
-				name: name
+	exports.get(name).then ->
```

Comment:
It should work, however using uuids/ids in the SDK makes it very hard to use the SDK and it'd be much nicer to handle readable names instead. 

I could port the SDK to uuids, but that'd also mean going one step back, since there were various discussions about needing the SDK to handle names as well.

I wonder why no one noticed this on time.